exercises/robot-name/RobotNameTest.cs

```diff
@@ -30,4 +31,20 @@ public void Can_reset_the_name()
         robot.Reset();
         Assert.NotEqual(originalName, robot.Name);
     }
+
+    [Fact(Skip = "Remove to run test")]
+    public void After_reset_the_name_is_valid() {
+        robot.Reset();
+        Assert.Matches(@"^[A-Z]{2}\d{3}$", robot.Name);
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Robot_names_are_unique() {
+        var names = new HashSet<string>();
+        for (int i = 0; i < 10_000; i++) {
+            Assert.False(names.Contains(robot.Name), $"Name {robot.Name} was already used ({i})");
+            names.Add(robot.Name);
+            robot.Reset();
```

Comment:
Instead of doing a reset, I think I would prefer to create a new robot in each iteration and then checking its name. This is also more in line with the test case's name.